test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs

```diff
@@ -588,9 +616,7 @@ private static string[] GetTestUrls(IServerAddressesFeature addressesFeature)
 
         private static string FixTestUrl(string url)
         {
-            var fixedUrl = url.Replace("://+", "://localhost")
-                .Replace("0.0.0.0", Dns.GetHostName())
-                .Replace("[::]", Dns.GetHostName());
```

Comment:
Did you try getting rid of `Replace("://+", "://localhost")` as well? It doesn't look like any test relies on it.